crates/fluvio-version-manager/src/command/switch.rs

```diff
@@ -0,0 +1,230 @@
+    fn copy_contents(&self, target: impl AsRef<Path>) -> Result<()> {
+        for entry in &self.contents {
+            let filename = entry.file_name().ok_or(anyhow::anyhow!(
+                "Failed to get filename from path: {}",
+                entry.display()
+            ))?;
+            let target_path = target.as_ref().join(filename);
+
+            copy(entry, &target_path)?;
+            tracing::info!(?target_path, "Copied binary");
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Parser)]
+pub struct SwitchOpt {
+    #[command(flatten)]
+    global_opts: GlobalOptions,
+    /// Version to set as active
+    #[arg(index = 1)]
+    version: Option<Channel>,
+}
+
+impl SwitchOpt {
+    pub async fn process(&self) -> Result<()> {
+        let Some(version) = self.version.clone() else {
+            self.notify_help(format!(
+                "You can use {} to see installed versions",
+                "fvm show".bold()
+            ));
+
+            return Err(anyhow::anyhow!("No version provided"));
+        };
+
+        let versions_path = fvm_versions_path()?;
+
+        // If the user runs this command before installing any versions, we
+        // should notify them and exit early.
+        if !versions_path.exists() {
+            self.notify_warn("No local Fluvio versions found.");
+            self.notify_help(format!(
+                "Try installing a version with {}, and then retry this command.",
+                "fvm install".bold()
+            ));
+
+            return Ok(());
+        }
+
+        let pkgset_path = versions_path.join(version.to_string());
+
+        // If the package is not available locally, we should notify the user
+        // and exit early.
+        if !pkgset_path.exists() {
+            self.notify_warn(format!(
+                "Fluvio version {} is not installed",
+                version.to_string().bold()
+            ));
+
+            let help = format!("fvm install {}", version);
+
+            self.notify_help(format!(
+                "Install the desired version using {}, and then retry this command.",
+                help.bold()
+            ));
+
+            return Ok(());
+        }
+
+        let fluvio_bin_dir = fluvio_binaries_path()?;
+
+        if !fluvio_bin_dir.exists() {
+            tracing::info!(?fluvio_bin_dir, "Creating fluvio binaries directory");
+            create_dir_all(&fluvio_bin_dir)?;
+        }
+
+        // Opens the version directory and then copies the contents (but the manifest)
+        // to the Fluvio Binaries directory
+        let version_dir = VersionDirectory::open(pkgset_path)?;
+        version_dir.copy_contents(&fluvio_bin_dir)?;
+
+        // Update settings file to keep track of active Fluvio Version
+        let manifest = VersionManifest::open(version_dir.manifest)?;
+        let mut settings = Settings::open()?;
+        settings.update_version(&manifest.channel, &manifest.version)?;
```

Comment:
Would prefer to move this code out of Cmd and into somewhere (maybe `VersionDirectory`) so it can be unit-tested indepently